capi_problems.rst

```diff
@@ -0,0 +1,428 @@
+technical and social. Different types of users bring different,
+sometimes conflicting, requirements. The tradeoff between stability
+and progress is an ongoing, highly contentious topic of discussion
+when suggestions are made for incremental improvements.
+Several proposals have been put forward for improvement, redesign
+or replacement of the C API, each representing a deep analysis of
+the problems.  At the 2023 Language Summit, three back-to-back
+sessions were devoted to different aspects of the C API. There is
+general agreement that a new design can remedy the problems that
+the C API has accumulated over the last 30 years, while at the same
+time updating it for use cases that it was not originally designed for.
+
+However, there was a also sense at the Language Summit that we are
+trying to discuss solutions without a clear common understanding
+of the problems that we are trying to solve. It was decided that
+we need to agree on the current problems with the C API, before
+we are able to evaluate any of the proposed solutions. We
+therefore created the
+[`capi-workgroup <https://github.com/capi-workgroup/problems/issues/>`__]
+repository on GitHub in order to collect everyone's ideas on that
+question.
+
+Over 60 different issues were created on that repository, each
+describing a problem with the C API. They were categorized and
+a number of recurring themes were identified. The sections below
+mostly correspond to these themes, and each contains a combined
+description of the issues raised in that category, along with
+links to the individual issues. In addition, we included a section
+that aims to identify the different stakeholders of the C API,
+and the particular requirements that each of them has.
+
+
+C API Stakeholders
+==================
+
+[TODO: write this section]
+
+API Evolution and Maintenance
+=============================
+
+The difficulty of making changes in the C API is central to this report. It is
+implicit in many of the issues we discuss here, particularly when we need to
+decide whether an incremental bugfix can resolve the issue, or whether it can
+only be addressed as part of an API redesign
+[`Issue 44 <https://github.com/capi-workgroup/problems/issues/44>`__]. The
+benefit of each incremental change is often viewed as too small to justify the
+disruption. Over time, this implies that every mistake we make in an API's
+design or implementation remains with us indefinitely.
+
+We can take two views on this issue. One is that this is a problem and the
+solution needs to be baked into any new C API we design, in the form of a
+process for incremental API evolution. The other possible approach is that
+this is not a problem to be solved, but rather a feature of any API. In this
+view, API evolution should not be incremental, but rather through large
+redesigns, each of which learns from the mistakes of the past and is not
+shackled by backwards compatibility requirements. A realistic approach
```

Comment:
This reads a bit too normative to me, inconsistent with the document's stated goal of listing only problems, not solutions.
```suggestion
shackled by backwards compatibility requirements. A compromise approach
```